sstable/Cargo.toml

```diff
@@ -7,6 +7,7 @@ license = "MIT"
 [dependencies]
 common = {path="../common", package="tantivy-common"}
 tantivy-fst = "0.4"
+zstd = "0.12"
```

Comment:
I am not quite clear whether this fits your MSRV, but workspace dependencies might be a good thing here as this is most likely not the first dependency on a specific pre-1.0 version of `zstd`, c.f. https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#inheriting-a-dependency-from-a-workspace